StuffAnThings/qbit_manage
# feat: add pysocks python module

## Description

This PR adds the [`pysocks`](https://pypi.org/project/PySocks/) python module so [Docker proxy config](https://docs.docker.com/network/proxy/) will work correctly with a socks5 proxy.

For example, adding the following env vars will cause qbit_manage to route all non-local network requests (on a `192.168.1.xx` subnet) via a socks5 proxy

```
https_proxy=socks5://proxy.example.com:3129
HTTPS_PROXY=socks5://proxy.example.com:3129
http_proxy=socks5://proxy.example.com:3128
HTTP_PROXY=socks5://proxy.example.com:3128
no_proxy=localhost,127.0.0.0/8,192.168.1.0/24
NO_PROXY=localhost,127.0.0.0/8,192.168.1.0/24
```

### To test

1. Build the Docker image locally
2. Run the container with the above env vars pointing at a valid socks5 proxy
3. Open a shell on the running container `docker exec -it qbit_manage bash`
4. Make sure you are getting back the public IP of the proxy using something like `curl icanhazip.com && curl https://icanhazip.com`

## Type of change

Please delete options that are not relevant.



## Checklist:

- [ ] I have added or updated the docstring for new or existing methods
- [x] I have modified this PR to merge to the develop branch
